src/limitless/limitless_monitor_service.cc

```diff
@@ -174,8 +177,66 @@ std::shared_ptr<HostInfo> LimitlessMonitorService::GetHostInfo(const std::string
     return host;
 }
 
-bool CheckLimitlessCluster(SQLHDBC hdbc) {
-    return LimitlessQueryHelper::CheckLimitlessCluster(hdbc);
+bool CheckLimitlessCluster(const SQLTCHAR *connection_string_c_str, const char *custom_errmsg_c_str, char *final_errmsg_c_str, size_t final_errmsg_size) {
+    SQLHENV henv = SQL_NULL_HANDLE;
+    SQLHDBC hdbc = SQL_NULL_HANDLE;
+
+    std::string custom_errmsg = custom_errmsg_c_str == nullptr ? "" : custom_errmsg_c_str;
+
+    if (!OdbcHelper::AllocateHandle(SQL_HANDLE_ENV, nullptr, henv, "Could not allocate environment handle during limitless check") ||
+        !OdbcHelper::SetHenvToOdbc3(henv, "Could not set henv to odbc3 during limitless check")) {
+        std::string errmsg = StringHelper::MergeStrings("Could not allocate environment handle.", custom_errmsg);
+
+        if (errmsg.size() >= final_errmsg_size) {
+            errmsg = StringHelper::MergeStrings("Warning: error string is truncated.", errmsg);
+        }
+
+        if (final_errmsg_c_str != nullptr) {
+            strncpy(final_errmsg_c_str, errmsg.c_str(), final_errmsg_size);
+        }
```

Comment:
Can be wrapped into a function. If its used in multiple places, it can become a function